packages/legacy-events/ReactSyntheticEventType.js

```diff
@@ -40,8 +40,11 @@ export type ReactSyntheticEvent = {|
     nativeEventTarget: EventTarget,
   ) => ReactSyntheticEvent,
   isPersistent: () => boolean,
-  _dispatchInstances: null | Array<Fiber | EventTarget> | Fiber | EventTarget,
-  _dispatchListeners: null | Array<Function> | Function,
+  isPropagationStopped: () => boolean,
+  _dispatchInstances: null | Array<Fiber | null>,
```

Comment:
OK, we're back to these being strict. This is a lot clearer IMO.

Now can we revert the changes to `executeDispatch` in `EventPluginUtils` from https://github.com/facebook/react/commit/a600408b284e92ed3d801d7f4e37d10b5da06b10?

I suggest:

```diff
- event.currentTarget =
-    inst.tag !== undefined ? getNodeFromInstance(inst) : inst;
+ event.currentTarget = getNodeFromInstance(inst);
```

I guess this isn't even used by the modern code path anymore.

Reply:
When did it stop being possible to have a single `Fiber` in there? Is it still possible in legacy system?

Reply:
In the legacy system it is, but not in the modern system. Given we've forked it, the old code should continue to work. I've re-added the type anyway, even if it's going away very soon.